### PR TITLE
Refactored project

### PR DESCRIPTION
**Pull request by obabec**
Finally refactored project

### CODE REVIEW

File: src/main/java/com/redhat/patriot/network_simulator/example/container/DockerCont.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                .withCmd()
+                .withName(name)
+                .exec();
+    }
+    public void connectContToNetwork(CreateContainerResponse cont, String networkId) {
+        dockerClient.connectToNetworkCmd().withNetworkId(networkId).withContainerId(cont.getId()).exec();
+
+    }
+
+    public void runCommand(   final CreateContainerResponse container,
```
Comment (mijaros) [inline src/main/java/com/redhat/patriot/network_simulator/example/container/DockerCont.java]: Please fix formatting here.

File: src/main/java/com/redhat/patriot/network_simulator/example/files/FileUtils.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.redhat.patriot.network_simulator.example.files;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.StandardCopyOption;
+
+public class FileUtils {
+    public String convertToFile(InputStream inputStream, String name) {
+        File targetFile = new File(name);
```
Comment (mijaros) [inline src/main/java/com/redhat/patriot/network_simulator/example/files/FileUtils.java]: And do you close this file? Wouldn't it be better to declare as resource in try block?

File: src/main/java/com/redhat/patriot/network_simulator/example/image/DockerImage.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    public DockerImage(DockerClient dockerClient) {
+        this.dockerClient = dockerClient;
+    }
+
+    public void buildImage(String tag, String path) {
+
+        FileUtils fileUtils = new FileUtils();
+
+        if (new File(DockerImage.class.getClassLoader().getResource(path).getFile()).exists()) {
```
Comment (mijaros) [inline src/main/java/com/redhat/patriot/network_simulator/example/image/DockerImage.java]: I'm not sure I understand this line, wouldn't is be better to save this file into some variable, you're opening it again later.

File: src/test/java/com/redhat/patriot/network_simulator/example/files/FileUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class FileUtilsTest {
+
+    @Test
+    void convertToFile() {
+
+        try {
+            File tmpDir = new File("tmpTestDir");
```
Comment (mijaros) [inline src/test/java/com/redhat/patriot/network_simulator/example/files/FileUtilsTest.java]: This is not correct way how to create temporary directory.
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createTempDirectory-java.lang.String-java.nio.file.attribute.FileAttribute...-

File: src/test/java/com/redhat/patriot/network_simulator/example/files/FileUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            e.printStackTrace();
+        }
+
+
+    }
+
+    @Test
+    void deleteDirWithFiles() {
+        try {
+            File tempDir = new File("tmpDir");
```
Comment (mijaros) [inline src/test/java/com/redhat/patriot/network_simulator/example/files/FileUtilsTest.java]: See above

File: src/test/java/com/redhat/patriot/network_simulator/example/files/FileUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    void writeToFile(String path) {
+        PrintWriter writer = null;
+        try {
+            writer = new PrintWriter(path, "UTF-8");
```
Comment (mijaros) [inline src/test/java/com/redhat/patriot/network_simulator/example/files/FileUtilsTest.java]: Use try with resources https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

File: src/main/java/com/redhat/patriot/network_simulator/example/files/FileUtils.java (hunk 2 above)
Comment (obabec) [reply src/main/java/com/redhat/patriot/network_simulator/example/files/FileUtils.java]: I think file java is only representing path, name,... File is not closable. I can add InputStream to resources, but I don't think it's necessary.

Comment (mijaros) [reply src/main/java/com/redhat/patriot/network_simulator/example/files/FileUtils.java]: Agreed.

File: pom.xml [hunk 7]
```diff
@@ -141,31 +144,76 @@
...
+         <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <configuration>
+               <source>8</source>
+               <target>8</target>
+            </configuration>
          </plugin>
+      <plugin>
+         <groupId>org.apache.maven.plugins</groupId>
```
Comment (mijaros) [inline pom.xml]: Isn't this redundant? See line 122

**Pull request merged**